webhook-channel-2023.bs

```diff
@@ -0,0 +1,190 @@
+Subscription Resource. The only required fields in the payload are *type*, *topic* and *sendTo*.
+* The **type** field MUST contain the type of channel being requested: `WebhookChannel2023`
+* The **topic** field MUST contain the URL of the resource a client wishes to subscribe to changes.
+* The **sendTo** field MUST contain the `https` URL of the webhook endpoint, where the notification sender
+    is expected to send notifications.
+
+For WebhookChannel2023 interactions, the subscription server responds with a subscription response.
+The only required fields in the payload are *id*, *type*, *topic*, *sender*
+* The **id** field MUST be a URI, which identifies created notification channel.
+* The **type** field MUST contain the type of channel created: `WebhookChannel2023` Web
+* The **topic** field MUST contain the URL of the resource which notifications will be about.
+* The **sender** field MUST contain URI, which identifies the notifications sender
+
+Issue(134):
+
+## Subscription Example ## {#example}
+
+*This section is non-normative.*
+
+An example `POST` request, including the webhook endpoint, using a `DPoP` bound access token is below:
+
+```http
+POST /subscription
+Host: channels.example
+Authorization: DPoP <token>
+DPoP: <proof>
+Content-Type: application/ld+json
+```
+```jsonld
+{
+  "@context": ["https://www.w3.org/ns/solid/notification/v1"],
+  "type": "WebhookChannel2023",
+  "topic": "https://storage.example/resource",
+  "sendTo": "https://webhook.example/871b84e7",
+  "state": "opaque-state",
+  "expiration": "2023-12-23T12:37:15Z",
+  "rate": "PT10s"
+}
+```
+> POST request including type, topic and sendTo targeting the Notification Subscription Resource.
+
+A successful response will contain a URL identifying the notification sender:
+
+```http
+Content-Type: application/ld+json
+```
+```jsonld
+{
+  "@context": "https://www.w3.org/ns/solid/notification/v1",
+  "id": "https://channels.example/ea1fcf13-7482-4bbb-a6c1-c168ddd7b0aa"
+  "type": "WebhookChannel2023",
+  "sender": "https://sender.example/#it",
+  "expiration": "2023-12-23T12:37:15Z",
+  "rate": "PT10s"
+}
+```
+> Response to the POST request, including channel id, type and the sender identity.
+
+# Authentication and Authorization # {#auth}
+
+* Subscription client MUST perform authenticated subscription request.
+* Notification Sender MUST perform authenticated request to *sendTo* webhook endpoint,
+    using identity provided as *sender* in the subscription response.
+
+As described by the Solid Notifications Protocol section on Authorization,
+the WebhookChannel2023 requires authorization and follows the guidance of the Solid Protocol
+sections on Authentication and Authorization [[!Solid.Protocol]].
+
+It is beyond the scope of this document to describe how a client fetches an access token.
+Solid-OIDC is one example of an authentication mechanism that could be used with Solid Notifications [[!Solid.OIDC]].
+
```

Comment:
I feel like this section is incomplete.

Pointing to Solid OIDC for the first use case (Subscription client MUST perform authenticated subscription request.) is fine as that use case is very clearly a client-pod relationship which is what Solid OIDC was designed to handle.

There are many questions left unanswered about the second use case (Notification Sender MUST perform authenticated request to *sendTo* webhook endpoint, using identity provided as *sender* in the subscription response.).

- Are we saying that a sender's "authenticated request" needs to be Solid OIDC as implied by the Solid Notifications Protocol?
- If so, what role is the sender? Is the sender the token issuer? Is the sender the token subject?
- In my opinion, Solid OIDC is an inappropriate Authentication protocol for this use case. It's designed for use cases that involve a user agent, and therefore has more complexities than are needed. This use case is a simple server-to-server communications, but the following unnecessary things would be required if we were forced to use Solid OIDC:
  - An identity provider, not for users, but the send itself
  - The use of DPoP tokens which are not needed because DPoP tokens are designed for ecosystems that have three constituents: Identity Provider, Relying Party, and Resource Server. 
 - I've been out of it for a bit. Perhaps Solid OIDC has added the Client Credentials flow. If so, this does solve the DPoP token problem mentioned about as Client Credentials doesn't use DPoP, but it still requires extra complexity because the the sender would need to authenticated with an identity provider before sending notifications.
 - I still prefer the method I outlined in the original document: Just give the sender its own public/private key pair and let it mint a simple token itself without needing to get a Solid OIDC IdP involved. Again, I've been out of it, so if there's an explanation as to why this approach is worse, I'm happy to hear it.
 - But, no matter what approach we go with, I would still prefer to have this section go into a little more detail on how senders send authenticated requests.

Reply:
@jaxoncreed I agreed with all your points, recently I [posted relevant comments on gitter](https://gitter.im/solid/authentication-panel?at=63d81b2b0c94855213d78674)
At the same time I don't think the problem is specific for WebHook channel type. I see it as more common requirement to authenticate server-side clients which we should be able to do in consistent way across the Solid ecosystem.

Reply:
I'm going to add #155 as inline issue. This way we can already start using Capability URL for *sendTo*, later adding support for Solid-OIDC and other more fitting options.

Reply:
![image](https://user-images.githubusercontent.com/876431/218345808-ee0a7d83-20dc-41ed-b291-d25ebd4a58b2.png)